### PR TITLE
UI Creation Flow

### PR DESCRIPTION
**Pull request by jonrossca**
Update the creation of the Host Connection, Server Connection, and Rooms List. Such that they are not created multiple times, and therefore not listening to signals from the client multiple times.

Vertically centered all headers. Moved connect button into the header.

### CODE REVIEW

File: client/css/client.css [hunk 1]
```diff
@@ -59,6 +62,12 @@ html, body {
   font-size: 1.0em;
 }
 
+.vertical_center_text {
+  position: absolute;
+	top: 50%;
+	transform: translateY(-50%);
```
Comment (flackr) [inline client/css/client.css]: nit: whitespace

File: client/css/client.css [hunk 2]
```diff
@@ -67,15 +76,19 @@ html, body {
...
 }
 
 .header {
+  position: relative;
   height: 64px;
   background: var(--primary-color);
   color: var(--light-text-color);
   font-size: 2.0em;
+/*  padding-left: 10px;
+  padding-top: 10px;*/
```
Comment (flackr) [inline client/css/client.css]: nit: remove commented code.

File: client/css/client.css [hunk 3]
```diff
@@ -137,30 +150,28 @@ html, body {
...
   visibility: hidden;
 }
 
+.settings_header_button {
+  position: absolute;
+	top: 50%;
+	transform: translateY(-50%);
+  right: 10px;
+  /*margin-right: 10px;
+  margin-top: 5px;*/
```
Comment (flackr) [inline client/css/client.css]: nit: Remove commented code.

File: client/js/client-ui.js [hunk 4]
```diff
@@ -240,14 +245,44 @@ class BaseUI {
...
+    }
   }
 
   onKeyPress(evt) {
-    //TODO parse irc commands here
     if (evt.keyCode == 13) {
       var elem = this.elem.querySelector('.main_input_text')
-      this.client.send(hostId, serverName, elem.value);
+      var text = elem.value;
+      if (this.input_index == this.input_history.length) {
```
Comment (flackr) [inline client/js/client-ui.js]: So if you arrow up to a previous command, send it, then you're at that position in the command history? Shouldn't it jump back to the bottom?

File: client/js/client-ui.js [hunk 5]
```diff
@@ -240,14 +245,44 @@ class BaseUI {
...
+        this.input_history.push(text);
```
Comment (flackr) [inline client/js/client-ui.js]: Let's add an aribitrary maximum number of commands, say 100? Make a constant at the top of the file for it like:
var MAX_INPUT_HISTORY = 100;

File: client/css/client.css (hunk 1 above)
Comment (jonrossca) [inline client/css/client.css]: odd, not showing locally. Redid the whitespace

File: client/css/client.css (hunk 2 above)
Comment (jonrossca) [inline client/css/client.css]: done

File: client/css/client.css (hunk 3 above)
Comment (jonrossca) [inline client/css/client.css]: done

File: client/js/client-ui.js (hunk 4 above)
Comment (jonrossca) [inline client/js/client-ui.js]: Touche, done.

File: client/js/client-ui.js (hunk 5 above)
Comment (jonrossca) [inline client/js/client-ui.js]: done

**Comment by jonrossca** [comment]
Updated based on review comments.

**Pull request merged**